### PR TITLE
DB batch transaction support, profiling, and misc updates

### PR DESCRIPTION
**Pull request by warptangent**
Add support to BlockchainDB and BlockchainLMDB for batch transactions.

Add profiling to block and tx processing and DB operations.

Improve block and tx processing efficiency by less repeat hashing.

Move LMDB storage to "lmdb" subfolder.
- Upon startup, if old LMDB files are detected, abort with a message for the user to move them to subfolder or delete them.

Update and fix log statements and formatting.

### CODE REVIEW

File: src/cryptonote_core/BlockchainDB_impl/db_lmdb.cpp [hunk 1]
```diff
@@ -672,7 +695,7 @@ void BlockchainLMDB::open(const std::string& filename)
   lmdb_db_open(txn, LMDB_OUTPUT_GINDICES, MDB_CREATE, m_output_gindices, "Failed to open db handle for m_output_gindices");
 *************************************************/
 
-  lmdb_db_open(txn, LMDB_SPENT_KEYS, MDB_CREATE, m_spent_keys, "Failed to open db handle for m_outputs");
+  lmdb_db_open(txn, LMDB_SPENT_KEYS, MDB_CREATE, m_spent_keys, "Failed to open db handle for m_spent_keys");
```
Comment (tewinget) [inline src/cryptonote_core/BlockchainDB_impl/db_lmdb.cpp]: Thought I changed those all with a vim macro...nice catch

File: src/cryptonote_core/BlockchainDB_impl/db_lmdb.cpp [hunk 2]
```diff
@@ -769,7 +805,7 @@ bool BlockchainLMDB::block_exists(const crypto::hash& h) const
   if (get_result == MDB_NOTFOUND)
   {
     txn.commit();
-    LOG_PRINT_L1("Block with hash " << epee::string_tools::pod_to_hex(h) << "not found in db");
+    LOG_PRINT_L1("Block with hash " << epee::string_tools::pod_to_hex(h) << " not found in db");
```
Comment (tewinget) [inline src/cryptonote_core/BlockchainDB_impl/db_lmdb.cpp]: Will probably change this to log level 2 or 3 as well, as it gets pretty spammy.

File: src/cryptonote_core/BlockchainDB_impl/db_lmdb.cpp [hunk 3]
```diff
@@ -1165,20 +1253,37 @@ uint64_t BlockchainLMDB::get_tx_block_height(const crypto::hash& h) const
   LOG_PRINT_L3("BlockchainLMDB::" << __func__);
   check_open();
 
+  // If m_batch_active is set, a batch transaction exists beyond this class,
```
Comment (tewinget) [inline src/cryptonote_core/BlockchainDB_impl/db_lmdb.cpp]: This block comment seems to apply class-wide, consider relocating?

Comment (warptangent) [inline src/cryptonote_core/BlockchainDB_impl/db_lmdb.cpp]: Agreed. I'll include in next update.

Comment (tewinget) [inline src/cryptonote_core/BlockchainDB_impl/db_lmdb.cpp]: Cool

File: src/cryptonote_core/BlockchainDB_impl/db_lmdb.cpp (hunk 2 above)
Comment (warptangent) [inline src/cryptonote_core/BlockchainDB_impl/db_lmdb.cpp]: Agreed. tx_exists() can similarly be updated.

**Pull request merged**